node/src/main/scala/org/bitcoins/node/networking/P2PClient.scala

```diff
@@ -196,13 +218,27 @@ case class P2PClientActor(
         //send them to 'context.parent' -- this is the
         //PeerMessageHandler that is responsible for
         //creating this Client Actor
-        messages.foreach { m =>
-          val msg = NetworkMessageReceived(m, P2PClient(self, peer))
-          peerMsgHandlerReceiver.handleNetworkMessageReceived(msg)
-
+        val newMsgReceiverF: Future[PeerMessageReceiver] = {
+          val currentF = Future.successful(currentPeerMsgHandlerRecv)
```

Comment:
This is hard to read... What's going on here? 

It looks similar to `FutureUtil.sequentially`, except that it has a custom initial value for the `foldLeft`. Would it be possible to add a method `FutureUtil.sequentiallyWithInit`, or something along those lines, and implement `FutureUtil.sequentially` as a special case of `FutureUtil.sequentiallyWithInit`?

Reply:
The goal here is to process the messages sequentially. 

To process the next message we need the previous message to process, and then we need to use the `PeerMessageReceiver` that is returned from the previously processed message.